types/node/index.d.ts

```diff
@@ -780,101 +787,60 @@ declare module "http" {
+        headers?: OutgoingHeaders;
         auth?: string;
         agent?: Agent | boolean;
         timeout?: number;
     }
 
-    export interface Server extends net.Server {
+    export class Server extends net.Server {
+        constructor(req: IncomingMessage, res: ServerResponse);
         setTimeout(msecs: number, callback: Function): void;
         maxHeadersCount: number;
         timeout: number;
         listening: boolean;
     }
-    /**
-     * @deprecated Use IncomingMessage
-     */
-    export interface ServerRequest extends IncomingMessage {
-        connection: net.Socket;
-    }
-
-    export interface ServerResponseHeaders {
-      [key: string]: number | string | string[];
-    }
-
-    export interface ServerResponse extends stream.Writable {
-        // Extended base methods
-        write(buffer: Buffer): boolean;
-        write(buffer: Buffer, cb?: Function): boolean;
-        write(str: string, cb?: Function): boolean;
-        write(str: string, encoding?: string, cb?: Function): boolean;
-        write(str: string, encoding?: string, fd?: string): boolean;
 
-        addTrailers(headers: ServerResponseHeaders): void;
+    export class OutgoingMessage extends stream.Writable {
         finished: boolean;
-        getHeader(name: string): number | string | string[] | undefined;
-        getHeaderNames(): string[]
-        getHeaders(): any
-        hasHeader(name: string): boolean
+        sendDate: boolean;
         headersSent: boolean;
+
+        constructor();
+        setTimeout(msecs: number, callback?: () => void): this;
+        setHeader(name: string, value: number | string | string[]): void;
+        getHeader(name: string): number | string | string[] | undefined;
         removeHeader(name: string): void;
-        sendDate: boolean;
-        setHeader(name: string, value: string | string[]): void;
-        setTimeout(msecs: number, callback: Function): ServerResponse;
+        addTrailers(headers: OutgoingHeaders): void;
+    }
+
+    export class ServerResponse extends OutgoingMessage {
         statusCode: number;
         statusMessage: string;
-        write(chunk: any, encoding?: string): any;
-        writeContinue(): void;
-        writeHead(statusCode: number, reasonPhrase?: string, headers?: ServerResponseHeaders): void;
-        writeHead(statusCode: number, headers?: ServerResponseHeaders): void;
 
-        // Extended base methods
-        end(): void;
-        end(buffer: Buffer, cb?: Function): void;
-        end(str: string, cb?: Function): void;
-        end(str: string, encoding?: string, cb?: Function): void;
-        end(data?: any, encoding?: string): void;
+        constructor(req: IncomingMessage);
+        writeContinue(cb?: () => void): void;
+        writeHead(statusCode: number, statusText?: string, headers?: OutgoingHeaders): void;
+        writeHead(statusCode: number, headers?: OutgoingHeaders): void;
+        assignSocket(socket: stream.Writable): void;
```

Comment:
Fair enough. Most of this is documented from my internal experience with the node types. I can add docblocks with `@internal` if that's the right way.